Shape_detection/examples/Shape_detection/region_growing_on_polyline_graph.cpp

```diff
@@ -0,0 +1,96 @@
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include "include/utils.h"
+
+// Typedefs.
+using Kernel  = CGAL::Exact_predicates_inexact_constructions_kernel;
+using Point_3 = typename Kernel::Point_3;
+
+using Surface_mesh = CGAL::Surface_mesh<Point_3>;
+using Face_range   = typename Surface_mesh::Face_range;
+using Edge_range   = typename Surface_mesh::Edge_range;
+
+using One_ring_query = CGAL::Shape_detection::Polygon_mesh::One_ring_neighbor_query<Surface_mesh>;
+using Plane_region   = CGAL::Shape_detection::Polygon_mesh::Least_squares_plane_fit_region<Kernel, Surface_mesh>;
+using RG_planes      = CGAL::Shape_detection::Region_growing<Face_range, One_ring_query, Plane_region>;
+
+using Vertex_to_point_map = typename Plane_region::Vertex_to_point_map;
+
+using Polyline_graph = CGAL::Shape_detection::Polygon_mesh::Polyline_graph<Kernel, Surface_mesh>;
+using Segment_range  = typename Polyline_graph::Segment_range;
+using Segment_map    = typename Polyline_graph::Segment_map;
+
+using Line_region  = CGAL::Shape_detection::Segment_set::Least_squares_line_fit_region<Kernel, Segment_range, Segment_map>;
+using Line_sorting = CGAL::Shape_detection::Segment_set::Least_squares_line_fit_sorting<Kernel, Segment_range, Polyline_graph, Segment_map>;
+using RG_lines     = CGAL::Shape_detection::Region_growing<Segment_range, Polyline_graph, Line_region, typename Line_sorting::Seed_map>;
+
+int main(int argc, char *argv[]) {
+
+  // Load data either from a local folder or a user-provided file.
+  const std::string input_filename = (argc > 1 ? argv[1] : "data/am.off");
+  std::ifstream in_off(input_filename);
+  std::ifstream in_ply(input_filename);
+  CGAL::set_ascii_mode(in_off);
+  CGAL::set_ascii_mode(in_ply);
+
+  Surface_mesh surface_mesh;
+  if (CGAL::read_OFF(in_off, surface_mesh)) { in_off.close();
+  } else if (CGAL::read_PLY(in_ply, surface_mesh)) { in_ply.close();
+  } else {
+    std::cerr << "ERROR: cannot read the input file!" << std::endl;
+    return EXIT_FAILURE;
+  }
+  const Face_range face_range = faces(surface_mesh);
+  const Edge_range edge_range = edges(surface_mesh);
+  std::cout << "* number of input faces: " << face_range.size() << std::endl;
+  std::cout << "* number of input edges: " << edge_range.size() << std::endl;
+  assert(face_range.size() == 7320);
+  assert(edge_range.size() == 10980);
+
+  const Vertex_to_point_map vertex_to_point_map(
+    get(CGAL::vertex_point, surface_mesh));
+
+  // Find planar regions.
+  One_ring_query one_ring_query(surface_mesh);
+  Plane_region plane_region(surface_mesh,
+    CGAL::parameters::all_default(), vertex_to_point_map);
+  RG_planes rg_planes(face_range, one_ring_query, plane_region);
+
+  std::vector< std::vector<std::size_t> > regions;
+  rg_planes.detect(std::back_inserter(regions));
+  std::cout << "* number of found planar regions: " << regions.size() << std::endl;
+  assert(regions.size() == 9);
```

Comment:
all asserts seem to be for a given input

Reply:
Yep, see above. I will remove them or make them general.